### ISSUE TITLE
Join on the same table twice. Make use of alias. 

### ISSUE THREAD

**Issue by lasseschmitt** [angel-dart/orm#31]
Giving the following example table

```
@serializable
@orm
class _Booking extends Model {
  DateTime dateTime;
  @BelongsTo(localKey: "pick_up_address_id")
  Address pickUpAddress;
  @BelongsTo(localKey: "drop_off_address_id")
  Address dropOffAddress;
}
```

This will creates a join statement like 

`SELECT bookings.id, bookings.pick_up_address_id, booking.drop_off_address_id, addresses.id, addresses.street, addresses.id, addresses.street, FROM "bookings" LEFT OUTER JOIN addresses ON bookings.pick_up_address_id = addresses.id LEFT OUTER JOIN addresses ON bookings.drop_off_address_id = addresses.id;`

Which causes an issue 

> Error executing SELECT statement. ERROR: table name "addresses" specified more than once

The solution would be to make use of aliases 

```
SELECT bookings.id, 
bookings.pick_up_address_id, 
bookings.drop_off_address_id, 
a1.id, a1.street,  
a2.id, a2.street, 
FROM "practice_lesson_bookings" 
INNER JOIN addresses a1 ON a1."id" = bookings.pick_up_address_id
INNER JOIN addresses a2 ON a2."id" = bookings.drop_off_address_id

```

**Comment by thosakwe** [comment]
Should I mark this as closed?

**Comment by lasseschmitt** [comment]
Solved: #32

**Comment by thosakwe** [comment]
Big changes are coming to this package very soon. [`package:angel_serialize`](https://github.com/angel-dart/serialize) is now in `2.0.0-alpha`, and the ORM too will have an overhaul soon.
